### DISCUSSION TITLE
why onMounted is invokes twice on SFC used on page?

### DISCUSSION THREAD

**Discussion by itsmnthn** [nuxt/nuxt#24184] [Questions] [answered]
I have a pages `pages/referral/[code].vue` and `pages/referral/index.vue`
The page code is simple `<template> <ReferralV2 :use-code="code" /> </template>`

The component ReferralV2 has few data fetching based on watcher and onMounted to display on template and getting debug log like this

![Screenshot 2023-11-08 at 1 46 10 PM](https://github.com/nuxt/nuxt/assets/25129674/8f2bd726-335b-4b94-a673-ac02a13154d7)

and switching between pages debug logs like this and this happens on every page

![image](https://github.com/nuxt/nuxt/assets/25129674/0ecba33b-347c-43f2-89bb-56a26c56ad2c)

![image](https://github.com/nuxt/nuxt/assets/25129674/b41b648b-dae3-4f6a-80a1-c4f5f4a61333)

Layout only mount once & this is what it looks like `<main class="flex flex-(col grow) py-2.50"> <slot /> </main>` 

Here is the reproduction
https://stackblitz.com/edit/github-fawa5w?file=layouts%2Fdefault.vue

**Comment by huang-julien** [comment]
Hey :wave: do you have a minimal reproduction please ? So we can see what's going on in the code

**Reply by itsmnthn** [reply]
Reproduction: https://stackblitz.com/edit/github-fawa5w?file=layouts%2Fdefault.vue

**Comment by itsmnthn** [comment, marked as answer]
#24187

use 

`<NuxtPage :page-key="route => route.fullPath" />`

instead of

`<NuxtPage :key="$route.fullPath" />`
